components/Buttons/SlideSelect/index.tsx

```diff
@@ -7,30 +7,45 @@ type TSSProps = {
     value: number;
     className?: string;
 } & Children;
-
 const SlideSelect: React.FC<TSSProps> = styled(({ onClick, value, children, className }: TSSProps) => {
     return (
-        <>
-            <div className={className}>
-                {React.Children.toArray(children).map((child, index) => {
-                    return (
-                        <SlideOption
-                            onClick={(e) => onClick(index, e)}
-                            key={`slide-option-${index}`}
-                            className={`${index === value ? 'selected' : ''}`}
-                        >
-                            {child}
-                        </SlideOption>
-                    );
-                })}
-            </div>
-        </>
+        <div className={className}>
+            {React.Children.toArray(children).map((child, index) => {
+                return (
+                    <SlideOption
+                        onClick={(e) => onClick(index, e)}
+                        key={`slide-option-${index}`}
+                        className={`${index === value ? 'selected' : ''}`}
+                    >
+                        {child}
+                    </SlideOption>
+                );
+            })}
+            <div className="bg" />
+        </div>
     );
 })`
     display: flex;
     margin: auto;
     border: 1px solid #3da8f5;
     border-radius: 20px;
+    height: 40px;
+    position: relative;
+
+    .bg {
+        transition: 0.5s;
+        background-color: #3da8f5;
+        height: 39px;
+        width: 170px;
+        border-radius: 18px;
+        position: absolute;
+        top: 0;
+    }
+
+    .selected + .bg {
```

Comment:
Would be good to make this work for all number of options eg if there is three options

Reply:
Other than that the animation looks nice, but maybe reduce the size of the selected bg to 50% of the width of the whole whing